src/internal/log/span.go

```diff
@@ -0,0 +1,138 @@
+package log
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// Level is the level at which to generate Span logs.
+type Level int
+
+const (
+	DebugLevel Level = 1
+	InfoLevel  Level = 2
+	ErrorLevel Level = 3
+)
+
+func (l Level) log(z *zap.Logger, msg string, fields ...Field) {
+	switch l { //exhaustive:enforce
+	case DebugLevel:
+		z.Debug(msg, fields...)
+		return
+	case InfoLevel:
+		z.Info(msg, fields...)
+		return
+	case ErrorLevel:
+		z.Error(msg, fields...)
+		return
+	}
+	z.DPanic("log: internal error: unknown level in Level.log call", zap.Int("level", int(l)))
+	z.Debug(msg, fields...)
+}
+
+// EndSpanFunc is a function that ends a span.
+type EndSpanFunc = func(fields ...Field)
+
```

Comment:
I notice that there are `ErrorL` (error and a level), `Errorp` (error pointer) and `ErrorpL` (error pointer and a level), but no error-without-a-level option, which is a bit surprising — wouldn’t that be the most commonly desired version?

Reply:
Yeah, it's built into zap, so zap.Error.  I'm debating whether or not to alias all of these into our own package or not.  It can't be this one because `Error` is already the logging function.  (Avoided in zap itself by making it a method on the *zap.Logger object.)